src/install-opentofu.ps1

```diff
@@ -0,0 +1,502 @@
+                    $cosignIdentity = "https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/tags/v${OPENTOFU_VERSION}"
+                } else {
+                    if ($opentofuVersion.Contains("alpha") -or $opentofuVersion.Contains("beta")) {
+                        $cosignIdentity="https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/heads/main"
+                    } else {
+                        $ver = [version]($opentofuVersion -replace "-alpha|-beta|-rc.*")
+                        $major = $ver.Major
+                        $minor = $ver.Minor
+                        $cosignIdentity="https://github.com/opentofu/opentofu/.github/workflows/release.yml@refs/heads/v${major}.${minor}"
+                    }
+                }
+            }
+
+            logInfo "Verifying signature against cosign identity ${cosignIdentity}..."
+            & $cosignPath verify-blob --certificate-identity $cosignIdentity --signature "${tempPath}/${sigFile}" --certificate "${tempPath}/${certFile}" --certificate-oidc-issuer $cosignOidcIssuer "${tempPath}/${sumsFile}"
+            if($?) {
+                logInfo "Signature verified."
+            } else {
+                throw [InstallFailedException]::new("Failed to verify ${opentofuVersion} with cosign. (${msg})")
+            }
+        }
+
+        $internalZipFile = Join-Path $tempPath $zipName
+
+        if ($allUsers -and (!
+            (New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent())).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator))
+        )
+        {
+            # Note on this section: we are requesting elevated privileges via UAC. Unfortunately, this is only possible
+            # by launching the current script again as Administrator. This can cause some weird parsing bugs in
+            # conjunction with Start-Process and the "powershell" command. (The parameter separation disappears.)
+            # Also note that when using Start-Process with RunAs, it is not possible to pass environment variables.
+            # (The documentation is lying!)
+            #
+            # TL;DR: Make sure to MANUALLY test privilege elevation if you change this as we can't test UAC in CI! Make
+            # sure to test with paths containing spaces too! Did I mention to test this? Test it. I'm serious.
+
+            logInfo "Unpacking with elevated privileges..."
+            $logDir = tempdir
+            # TODO capture the log output of the shell running as admin and clean up afterwards.
+            $argList = @("-NonInteractive", "-File", ($scriptCommand | escapePathArgument), "-internalContinue", "-allUsers", "-installMethod", "standalone", "-installPath", ($installPath | escapePathArgument), "-internalZipFile", ($internalZipFile | escapePathArgument))
```

Comment:
Since this is deep inside the standalone installer this is the only method that makes sense. When we add more installation methods we will need to refactor this and provide a generic UAC privilege elevation method.